obsidianirc.c

```diff
@@ -0,0 +1,640 @@
+/* Copyright © 2025 Valware & ObsidianIRC Team
+ * License: GPLv3
+ * Name: third/obsidian-register
+ */
+/*** <<<MODULE MANAGER START>>>
+module
+{
+        documentation "https://github.com/link";
+        troubleshooting "In case of problems, check the documentation or e-mail me at v.a.pond@outlook.com";
+        min-unrealircd-version "6.1.0";
+        max-unrealircd-version "6.*";
+        post-install-text
+        {
+                "The module is installed. Now all you need to do is add a loadmodule line:";
+                "loadmodule \"third/obsidian-register\";";
+                "The module needs no other configuration.";
+                "Once you're good to go, you can finally type in your shell: ./unrealircd rehash";
+        }
+}
+*** <<<MODULE MANAGER END>>>
+*/
+
+/* One include for all */
+#include "obsidian.h"
+
+long CAP_ACCOUNTREGISTRATION = 0L;
+
+ModuleHeader MOD_HEADER
+= 
+{
+    "third/obsidianirc", /* Name of module */
+    "1.0", /* Version */
+    "ObsidianIRC", /* Short description of module */
+    "ObsidianIRC Team", /* Author */
+    "unrealircd-6", /* Version of UnrealIRCd */
+};
+
+/**
+ * MOD_INIT - Module initialization routine.
+ */
+MOD_INIT()
+{
+    ModDataInfo mreq;
+
+    memset(&mreq, 0, sizeof(mreq));
+    mreq.name = "sasl_auth_type";
+    mreq.free = sat_free;
+    mreq.serialize = sat_serialize;
+    mreq.unserialize = sat_unserialize;
+    mreq.type = MODDATATYPE_CLIENT;
+    if (!(sasl_md = ModDataAdd(modinfo->handle, mreq)))
+    {
+        config_error("Could not add ModData for sasl_auth_type. Please contact developer.");
+        return MOD_FAILED;
+    }
+
+    if (open_database(OBSIDIAN_DB) != SQLITE_OK)
+    {
+        config_error("Could not open database. Please contact ObsidianIRC Support.");
```

Comment:
Would really not ask to contact support with only that info :D maybe we can find out more about why it failed. Same for all error handling here, we should at leats help ourselves to find out what is wrong easier from these messages.

Reply:
Not sure on this, please contact ObsidianIRC Support for more information.